pkg/runner/cluster_swarm.go

```diff
@@ -162,8 +162,14 @@ func (*ClusterSwarmRunner) Run(input *api.RunInput) (*api.RunOutput, error) {
 			// if we are keeping the service, we must also keep the network.
 			return
 		}
-		if err := cli.NetworkRemove(ctx, networkID); err != nil {
```

Comment:
That's something I lost a bit of time trying to figure it out - if we don't explicitly define a `ctx` within the scope of this `defer` statement, we actually pick up the `global ctx` defined in the beginning of the function, which gets canceled by the time we call `cli.NetworkRemove`, so the `cli.NetworkRemove` is never successful - because we try to execute it with an already canceled context.